File: Server/src/main/java/lk/wnl/wijeya/entity/SupplierPaymentGrn.java
package lk.wnl.wijeya.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Getter
@Setter
@Entity
@Table(name = "supplier_payment_grn")
public class SupplierPaymentGrn {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;
    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "supplier_payment_id", nullable = false)
    private SupplierPayment supplierPayment;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "paper_grn_id")
    private PaperGrn paperGrn;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "material_grn_id")
    private MaterialGrn materialGrn;

}

File: Client/src/app/entity/supplierpaymentgrn.ts
import {Supplierpayment} from "./supplierpayment";
import {PaperGrn} from "./PaperGrn";
import {MaterialGrn} from "./MaterialGrn";

export class Supplierpaymentgrn{
  id?: number;
  supplierPaymentDto?: Supplierpayment;
  paperGrn?: PaperGrn|null;
  materialGrn?: MaterialGrn|null;


  constructor(paperGrn: PaperGrn|null, materialGrn: MaterialGrn|null) {
    this.paperGrn = paperGrn;
    this.materialGrn = materialGrn;
  }
}
